### DISCUSSION TITLE
Gradient struggle : "bg-gradient-to-r class does not exist"

### DISCUSSION THREAD

**Discussion by bboy-space** [tailwindlabs/tailwindcss#6407] [Help] [answered]
Hello everyone, 

I'm trying to upgrade Tailwind from v2 to v3 and i'm struggling with the gradients. From the official doc (https://tailwindcss.com/docs/background-image ) I can find an example with bg-gradient-to-r from-cyan-500 to-blue-500 which work pretty fine directly as classes in the HTML . BUT I want to wrap this in a custom class to save time in my dev with something like :
 
`  .bg-gradient {
    @apply bg-gradient-to-r from-blue-500 to-blue-50  text-black  font-bold;
  }
`

At compilation time I have the following error :

> **The bg-gradient-to-r class does not exist. If bg-gradient-to-r is a custom class, make sure it is defined within a @layer directive.**

I tried the very basic example below in the playground and have the same result. 
 
@tailwind base;
@tailwind components;
@layer components {
  .bg-gradient {
    @apply bg-gradient-to-r from-blue-500 to-red-500;
  }
}
@tailwind utilities;

So is there something I missed or misunderstood in the concept of tailwind ?? 
Thank you very much for your help guys .

**Comment by bboy-space** [comment, marked as answer]
I fixed my problem. Simply by renaming my custom class !
Starting with "bg-gradient" was for sure confusing in the tailwind engine.

**Comment by sergiofalcon** [comment]
Same here

Rebuilding...
CssSyntaxError: tailwindcss:styles.css:23:9: The `bg-gradient-to-br` class does not exist. If `bg-gradient-to-br` is a custom class, make sure it is defined within a `@layer` directive.

Freshly install of TailwindCSS. In other projects, no problem at all. If i remove the gradient, everything works fine. 

Tried with dozens of class names, same result.

Only occurs with bg-gradient-to-? classes.

**Reply by tlhfckoctbcr** [reply]
May or may not be related, but I did a fresh install of PostCSS, autoprefixer, Tailwind. I get a similar error with `scale-{{ value }}` classes. 
`The 'scale-100' class does not exist. If 'scale-100' is a custom class, make sure it is defined within a '@layer' directive.`
